### DISCUSSION TITLE
html style changes using <Link> same as <a> element

### DISCUSSION THREAD

**Discussion by robert-dm** [vercel/next.js#14776] [Help] [answered]
# Bug report

## Describe the bug
The html element was changed to yellow background as indicated here: https://nextjs.org/learn/basics/navigate-between-pages/client-side but when I navigate to first-post.js page, the background changes from yellow (index) to white (first-post), but example indicates it should not change the color.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to the learn section and follow steps https://nextjs.org/learn/basics/navigate-between-pages/client-side 
2. Click on first-page hyperlink created with <Link>, after updated the html style to background: yellow
3. background is white

## Expected behavior
first-page.js background color should be yellow

## Screenshots
![image](https://user-images.githubusercontent.com/35821871/86275170-3e75dc00-bba9-11ea-8c65-40cc08f6f9aa.png)
![image](https://user-images.githubusercontent.com/35821871/86275187-459cea00-bba9-11ea-9110-739c0639ded5.png)

## System information

- OS: macOS Catalina
- Browser: both Chrome and Safari
- Version of Next.js: not sure how to check my next.js version but I created app following Learn steps.
- Version of Node.js: 12.18.0

## Additional context
<style jsx global>{`
        html,
        body {
          padding: 0;
          margin: 0;
          background: yellow;
          font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto,
            Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue,
            sans-serif;
        }

        * {
          box-sizing: border-box;
        }
      `}</style>

**Comment by jamesmosier** [comment, marked as answer]
Could you create a repository showing the failing case? I'm having trouble reproducing this based on your current steps.

Also in your reproduction, it seems to be missing a word:

> Click on first-page hyperlink created with , after updated the html style to background: yellow

created with what?

**Reply by robert-dm** [reply]
Hi @jamesmosier thanks for your reply.  You can check the repo here https://github.com/robert-dm/nextjs-blog-2, and the vercel link here: https://nextjs-blog-2-blue.vercel.app/ where you can change the background to yellow and you will see when you click in the "Read this page!" hyperlink will make the yellow color disappear when accessing to the post url https://nextjs-blog-2-blue.vercel.app/posts/first-post

In my previous comment, I missed the word <Link>, I mean to say the first-page hyperlink was created with <Link> as states in the tutorial.

Let me please know if you can reproduce it.

Thanks!

**Reply by robert-dm** [reply]
I have uploaded a video recording showing the behavior here: https://github.com/robert-dm/nextjs-blog-2/blob/master/recording.mov

Please note this is an app created using the below sentences as showed in NextJS tutorial:
npm init next-app nextjs-blog --example "https://github.com/vercel/next-learn-starter/tree/master/assets-metadata-css-starter"

Just by doing that, with no changes to the code, I have reproduced this issue.

Thanks.

**Reply by jamesmosier** [reply]
So it actually is doing a client side route, the issue is you are placing the CSS style on an element that gets recreated when the next page loads so your custom style gets overwritten.

There are 2 ways to test this:

1. Add the `background: yellow;` as a completely new style to the `body` 
<img width="1140" alt="Screen Shot 2020-07-14 at 7 30 48 PM" src="https://user-images.githubusercontent.com/2854919/87486128-8a446e80-c608-11ea-9f32-672ee595573e.png">
2. Check out your Network tab in Chrome Devtools. You will see that as soon as you hover over the link, it will prefetch the first-post.js file and then when you click the link no other resources will load in the Network panel.
<img width="694" alt="Screen Shot 2020-07-14 at 7 31 17 PM" src="https://user-images.githubusercontent.com/2854919/87486162-a9db9700-c608-11ea-95cf-5e2d9343966f.png">

**Reply by robert-dm** [reply]
Thank you!

**Reply by HarleyGolfGuy** [reply]
@jamesmosier the tutorial specifically says:  

> Use the browser’s developer tools to change the \'background\' CSS property of \<html\> to \'yellow\'.

So, is the tutorial incorrect?  If so, please have someone at next.js fix the tutorial.

**Reply by HarleyGolfGuy** [reply]
@jamesmosier also, when I hover over the link I don't see anything prefetched.

![image](https://user-images.githubusercontent.com/4811987/143936648-eab050cc-b658-4631-b6cc-a083a677297b.png)

**Reply by tomashaddad** [reply]
@HarleyGolfGuy According to [the documentation on \<Link /\>](https://nextjs.org/docs/api-reference/next/link), prefetching is only available in production. So, you will need to run `npm run build` and then `npm run start` to see prefetching in action. By default, pages are prefetched when they enter the viewport. To see the prefetch occur in the network dev tools, you need to have it open for it to capture the fetch, in which case both the `index.js` and `first-post.js` files will be listed. If you add the `prefetch={false}` prop to both links, you can see them prefetch on hover.

**Comment by bonellia** [comment]
Thank you very much, I am so glad this was asked by @robert-dm and addressed by @jamesmosier , because I recently started following the guide and honestly, I was going to be disappointed/discouraged from the beginning, not because it is a huge deal but I get worried when documentations have this kind of issues early on.
